File: src/main/java/AstNodes/FunctionCallAst.java
package AstNodes;

import java.util.List;

public class FunctionCallAst extends BaseAst{
    private String functionName;
    private List<BaseAst> params;

    public String getFunctionName() {
        return functionName;
    }

    public void setFunctionName(String functionName) {
        this.functionName = functionName;
    }

    public FunctionCallAst(String functionName, List<BaseAst> params) {
        this.functionName = functionName;
        this.params = params;
    }
}

File: src/main/java/ASTNodes/FunctionCallAST.kt
package ASTNodes

class FunctionCallAST(var functionName: String?, var params: List<BaseAST>?) : BaseAST() {

    override fun accept(visitor: ASTVisitor) {
        for (ast in params!!) {
            ast.accept(visitor)
        }
        visitor.visit(this)
    }
}
